Console/Command/BatchingOptimizeCommand.php

```diff
@@ -0,0 +1,366 @@
+
+        $recommendedBatchCountLow = $this->getRecommendedBatchCount($sizeAverage, $standardDeviation, self::INCREASED_MARGIN);
+        $recommendedBatchCountHigh = $this->getRecommendedBatchCount($sizeAverage, $standardDeviation);
+        $this->output->writeln('<info> ============ </info>');
+        $this->output->writeln('<info>Recommended batch count (low)</info>  : ' . $recommendedBatchCountLow . ' records');
+        $this->output->writeln('<info>Recommended batch count (high)</info> : ' . $recommendedBatchCountHigh . ' records');
+        $this->output->writeln(' ');
+        $this->output->writeln(
+            'This will override your "Maximum number of records processed per indexing job" configuration to <info>' . $recommendedBatchCountLow . '</info> for store "' . $storeName . '".');
+
+        if ($this->confirmOperation()) {
+            $this->configWriter->save(
+                ConfigHelper::NUMBER_OF_ELEMENT_BY_PAGE,
+                $recommendedBatchCountLow,
+                'stores',
+                $storeId
+            );
+        }
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     * @throws AlgoliaException
+     * @throws DiagnosticsException
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    protected function setStoreCounts(int $storeId): void
+    {
+        $simpleProducts = $this->getProductsCollectionForStore($storeId, self::PRODUCTS_SIMPLE_TYPES);
+        $complexProducts = $this->getProductsCollectionForStore($storeId, self::PRODUCTS_COMPLEX_TYPES);
+
+        $this->storeCounts[$storeId] = [
+            'simple' => $simpleProducts->count(),
+            'complex' => $complexProducts->count()
+        ];
+
+        $this->storeCounts[$storeId]['total'] =
+            (int) $this->storeCounts[$storeId]['simple'] + (int) $this->storeCounts[$storeId]['complex'];
+
+        $this->storeCounts[$storeId]['simple_percentage'] =
+            ($this->storeCounts[$storeId]['simple'] * 100) / $this->storeCounts[$storeId]['total'];
+
+        $this->storeCounts[$storeId]['complex_percentage'] =
+            ($this->storeCounts[$storeId]['complex'] * 100) / $this->storeCounts[$storeId]['total'];
+
+        $sampleSize = $this->input->getOption(self::LARGE_SAMPLE_OPTION) ?
+            self::LARGE_SAMPLE_SIZE :
+            self::DEFAULT_SAMPLE_SIZE;
+        $simpleSampleSize = (int)round($sampleSize * ($this->storeCounts[$storeId]['simple_percentage'] / 100));
+        $complexSampleSize = (int)round($sampleSize * ($this->storeCounts[$storeId]['complex_percentage'] / 100));
+
+        $this->storeCounts[$storeId]['simple_sample_size'] = $simpleSampleSize;
+        $this->storeCounts[$storeId]['complex_sample_size'] = $complexSampleSize;
+
+        $this->storeCounts[$storeId]['sample'] = array_merge(
+            $this->getProductsSizes($simpleProducts, $simpleSampleSize),
+            $this->getProductsSizes($complexProducts, $complexSampleSize)
+        );
```

Comment:
Could we add a max product object size somehwere here? This stat could serve to inform the absolute floor for batching (for a worst case scenario).